src/pages/index.js

```diff
@@ -15,80 +15,194 @@ import Section from "../components/Section";
+  userData = userInfo.getUserInfo();
   editUserInfoPopup.setInputValues({
-    title: userInfo.name,
-    description: userInfo.job,
+    title: userData.name,
+    description: userData.job,
   });
   formValidators["profile-form"].resetValidation();
   editUserInfoPopup.open();
 });
 
+editAvatarButton.addEventListener("click", () => {
+  userData = userInfo.getUserInfo();
+  editAvatarPopup.setInputValues({
+    link: userData.src,
+  });
+  formValidators["avatar-form"].toggleButtonState();
+  editAvatarPopup.open();
+});
+
 // Functions
 function handleImageClick(link, name) {
   imagePreviewPopup.open(link, name);
 }
 
 function handleAddCardSubmit(cardData) {
-  cardSection.addItem(
-    createCard({ name: cardData.title, link: cardData.link })
-  );
+  newCardPopup.setButtonText(true);
+  api
+    .addNewCard(cardData.title, cardData.link)
+    .then((data) => {
+      cardSection.addItem(createCard(data));
+    })
+    .catch((err) => {
+      console.error(`There is an Error: `, err); // log the error to the console
+    })
+    .finally(() => {
+      newCardPopup.setButtonText(false);
+    });
+
   newCardPopup.close();
 }
 
 function handleProfileEditSubmit(cardData) {
-  editUserInfo.setUserInfo(cardData.title, cardData.description);
+  editUserInfoPopup.setButtonText(true);
+  api
+    .editUserInfo(cardData.title, cardData.description)
+    .then(() => {
+      userInfo.setUserInfo(cardData.title, cardData.description);
+    })
+    .catch((err) => {
+      console.error(`There is an Error: `, err); // log the error to the console
+    })
+    .finally(() => {
+      editUserInfoPopup.setButtonText(false);
+    });
   editUserInfoPopup.close();
 }
 
+function handleCardDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setDeleteText(true);
+  deleteCardPopup.setConfirmDelete(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.removeCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.error(`There is an Error: `, err); // log the error to the console
+      })
+      .finally(() => {
+        deleteCardPopup.setDeleteText(false);
+      });
+  });
+}
+
+function handleAvatarEditSubmit(data) {
+  editAvatarPopup.setButtonText(true);
+  api
+    .updateAvatar(data.link)
+    .then(() => {
+      userInfo.setUserAvatar(data.link);
+    })
+    .catch((err) => {
+      console.error(`Avatar not updated: `, err); // log the error to the console
+    })
+    .finally(() => {
+      editAvatarPopup.setButtonText(false);
+    });
+
+  editAvatarPopup.close();
+}
+
 function createCard(cardData) {
   const cardElement = new Card(
     cardData,
     selectors.cardTemplate,
-    handleImageClick
+    handleImageClick,
+    handleCardDeleteClick,
+    handleCardLike
   );
   return cardElement.getCard();
 }
 
+function handleCardLike(data) {
+  // console.log(data);
+  api
+    .updateLike(data._id, data._isLiked)
```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.

Reply:
_isLiked isn't a method. I'm grabbing the isLiked property of data I thought?

Reply:
> _isLiked isn't a method. I'm grabbing the isLiked property of data I thought?

Please, delete the **low dash** (`underscore`) from methods/fields